internal/filter/filterer.go

```diff
@@ -0,0 +1,68 @@
+package filter
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+)
+
+type Filterer interface {
+	Filter([]string) ([]string, error)
+}
+
+type FiltererMode int
+
+const (
+	BlackList FiltererMode = iota
+	WhiteList
+)
```

Comment:
I was wondering if there is a good way of verifying that only those enum values will be used, so that I don't need to check in runtime if passed FiltererMode is one of [BlackList, WhiteList]. One of the approaches would be the one described here in _**slugs**_ section https://threedots.tech/post/safer-enums-in-go/. Is this approach common?

Reply:
Yes, commonly used approach